### PR TITLE
Refactoring CPRI-4159: work URN Validation Service

### PR DESCRIPTION
**Pull request by eshunsharma**
Story : https://agile-jira.pearson.com/browse/CPRI-3793

Sub Task: https://agile-jira.pearson.com/browse/CPRI-4159

### CODE REVIEW

File: src/services/assetRequest/__test__/searchWorkByURN.test.js [hunk 1]
```diff
@@ -27,26 +28,30 @@ describe('searchBySourceWorkService', () => {
...
     }
 
-    await searchBySourceWork(config, sourceWorkUuids)
+    await searchWorkByURN(config, sourceWorkUuids)
 
-    mdsSearchStub.should.have.been.calledWith(searchBySourceWorkGraph, config, payload)
+    mdsSearchStub.should.have.been.calledWith(searchWorkByURNGraph, config, payload)
   })
 
   it('should return false if sourceWorkUuids are not provided', async () => {
```
Comment (Kerenhb) [inline src/services/assetRequest/__test__/searchWorkByURN.test.js]: sourceWorkUuids -> work URNs

File: src/services/assetRequest/searchWorkByURN.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }, {})
+
+  const result = data.reduce((accumulator, currentValue) => {
+    if (isSourceWork(currentValue.type)) {
+      return { ...accumulator, [currentValue.id]: 'Asset not found for Source URN.' }
+    }
+
+    return { ...accumulator, [currentValue.id]: currentValue }
+  }, workURNsObject)
+
```
Comment (Kerenhb) [inline src/services/assetRequest/searchWorkByURN.js]: You loop though all this twice, you should be able to loop though the workURNsObject and map each one to one of the three cases right? I guess you would need to search data each time.

Comment (Kerenhb) [reply src/services/assetRequest/searchWorkByURN.js]: I guess it's not a big deal though, you want to return an object which makes it a bit trickier.

File: src/utils/index.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { mapYesNoField } from './mapYesNoField'
 import checkHeaderError from './checkHeaderError'
 import checkAllFieldsEmpty from './checkAllFieldsEmpty'
 import { removeEmptyFields } from './removeEmptyFields'
+import { isSourceWork } from './typeInspector'
```
Comment (Kerenhb) [inline src/utils/index.js]: I guess we should export all the functions in typeInspector out here, maybe even refactor where there used.
Should be minor

File: src/services/assetRequest/__test__/searchWorkByURN.test.js [hunk 4]
```diff
@@ -2,14 +2,15 @@ import { expect } from 'chai'
...
 // eslint-disable-next-line import/named
-import { searchBySourceWork, __RewireAPI__ as rewireAPI } from '../searchBySourceWork'
-import { searchBySourceWorkGraph } from '../graph/searchbySourceWork'
+import { searchWorkByURN, __RewireAPI__ as rewireAPI } from '../searchWorkByURN'
+import { searchWorkByURNGraph } from '../graph/searchWorkByURN'
 
 const expectedOutput = {
   data: [{
-    id: '12345'
+    id: '456'
```
Comment (Kerenhb) [inline src/services/assetRequest/__test__/searchWorkByURN.test.js]: Add type: ['Work'] for completness

File: src/services/assetRequest/__test__/searchWorkByURN.test.js [hunk 5]
```diff
@@ -2,14 +2,15 @@ import { expect } from 'chai'
...
   }, {
-    id: '123'
+    id: '123',
+    type: ['SourceWork']
```
Comment (Kerenhb) [inline src/services/assetRequest/__test__/searchWorkByURN.test.js]: Could expand it to  type: ['Work', 'SourceWork'] to check it works correctly

File: src/utils/typeInspector.js [hunk 6]
```diff
@@ -1,2 +1,3 @@
 export const isTextQuotation = (type = []) => type.includes('TextSourceQuotation')
 export const isMediaQuotation = (type = []) => type.includes('MediaSourceQuotation')
+export const isSourceWork = (type = []) => type.includes('SourceWork')
```
Comment (Kerenhb) [inline src/utils/typeInspector.js]: Add a test for this

File: src/services/assetRequest/searchWorkByURN.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  const { data } = await mdsSearch(searchWorkByURNGraph, config, payload)
+
+  const workURNsObject = workURNs.reduce((accumulator, currentValue) => {
+    return { ...accumulator, [currentValue]: 'Asset not found.' }
+  }, {})
+
+  const result = data.reduce((accumulator, currentValue) => {
+    if (isSourceWork(currentValue.type)) {
+      return { ...accumulator, [currentValue.id]: 'Asset not found for Source URN.' }
+    }
```
Comment (Kerenhb) [inline src/services/assetRequest/searchWorkByURN.js]: Could set  'Asset not found for Source URN' to a variable then use it in the return statement below, something like
```
   (accumulator, currentValue) => {
   let value = currentValue
    if (isSourceWork(currentValue.type)) value = 'Asset not found for Source URN.' 
    return { ...accumulator, [currentValue.id]: value }
```

Comment (eshunsharma) [reply src/services/assetRequest/searchWorkByURN.js]: I think this is better:
```
const value = isSourceWork(currentValue.type) ? 'Asset not found for Source URN.' : currentValue

return { ...accumulator, [currentValue.id]: value }

```
No re assigning of value.

**Pull request merged**